silhouette/app/com/mohiva/play/silhouette/api/services/AuthenticatorService.scala

```diff
@@ -128,7 +150,7 @@ trait AuthenticatorService[T <: Authenticator] {
    * @param authenticator The authenticator to touch.
    * @return The touched authenticator on the left or the untouched authenticator on the right.
    */
-  protected[silhouette] def touch(authenticator: T): Either[T, T]
+  def touch(authenticator: T): Either[T, T]
```

Comment:
I dont get the 'more fluent interface' because there is a `implicit class` to convert Either[T,T] to T

```scala
private implicit class ExtractEither[T](r: Either[T, T]) {
	def extract: T = r.fold(identity, identity)
}
private def handleInitializedAuthenticator[T](authenticator: A, block: A => Future[HandlerResult[T]])(implicit request: RequestHeader) = {
	val auth: Either[T, T] = env.authenticatorService.touch(authenticator)
    block(auth.extract).flatMap {
    	...
    }
}
```

but if it uses a option as return that can be done without implicit convert.

```scala
private def handleInitializedAuthenticator[T](authenticator: A, block: A => Future[HandlerResult[T]])(implicit request: RequestHeader) = {
	val auth: Option[T] = env.authenticatorService.touch(authenticator)
    block(auth.getOrElse(authenticator)).flatMap {
    	...
    }
}
```
I always prefer a code without implicit that's more clean to understand